src/sentry/integrations/perforce/client.py

```diff
@@ -308,9 +355,85 @@ def get_blame_for_files(
         Note: This does not provide line-specific blame. It returns the most recent
         changelist for the entire file, which is sufficient for suspect commit detection.
 
+        API docs:
+        - p4 filelog: https://www.perforce.com/manuals/cmdref/Content/CmdRef/p4_filelog.html
+        - p4 describe: https://www.perforce.com/manuals/cmdref/Content/CmdRef/p4_describe.html
+
         Returns a list of FileBlameInfo objects containing commit details for each file.
         """
-        return []
+        metrics.incr("integrations.perforce.get_blame_for_files")
+        blames = []
+        p4 = self._connect()
+
+        # Cache user info to avoid multiple lookups for the same user
+        user_cache: dict[str, dict[str, Any] | None] = {}
+
+        try:
+            for file in files:
```

Comment:
Could there be rate limit concerns?  We're contacting the api like ~3 times per file for X files, I know for some other source code integrations we have some logic to handle rate limits so idk if it's applicable here.